FlowCrypt/Controllers/Compose/ComposeViewController.swift

```diff
@@ -1074,11 +1069,34 @@ extension ComposeViewController {
 
 // MARK: - Action Handling
 extension ComposeViewController {
+    @MainActor
+    func encryptAndSend(message: SendableMsg, threadId: String?) async throws {
+        try await composeMessageService.encryptAndSend(
+            message: message,
+            threadId: threadId
+        )
+    }
+
+    @MainActor
+    func searchContacts(query: String) async throws -> [String] {
+        return try await cloudContactProvider.searchContacts(query: query)
+    }
+
+    @MainActor
+    func findLocalContact(with email: String) async throws -> RecipientWithSortedPubKeys? {
+        return try await localContactsProvider.searchRecipient(with: email)
+    }
+
+    @MainActor
+    func fetchContact(with email: String) async throws -> RecipientWithSortedPubKeys {
+        return try await pubLookup.fetchRemoteUpdateLocal(with: email)
+    }
```

Comment:
Sorry - I misdirected you. These should be part of an actor, yes. But not MainActor. These are all non-main thread tasks (making an API call, encrypting something etc - all are heavy tasks that should not be on main).

At a first glance, that would suggest that `localContactsProvider` and `cloudContactProvider` and `pubLookup` and `composeMessageService` should be actors. But I don't see any benefit in that? Actor will ensure that only one method runs concurrently. But there is no limitation on making concurrent API calls - no harm in not being implemented as an actor.

Therefore, please try another change - remove these 4 new `@MainActor` methods, and use their contents directly within this class. Let's see what happens.

Reply:
@tomholub Yeah, I agree with you. Those tasks should not run on main thread (Sorry, i misunderstood you.)